### PR TITLE
perf(autoware_ndt_scan_matcher): remove evecs_, evals_ of Leaf for memory efficiency

### PR DESCRIPTION
**Pull request by taisa1**
## Description
By investigating the memory usage during scan matching, I found that the memory consumption was dominated by `std::vector<MultiVoxelGridCovariance<pcl::PointXYZ>::Leaf>` in addition to the map itself. 

The following figure shows the memory usage when executing autoware's `ndt_scan_matcher` with `sample-map-rosbag`. The bottom area corresponds to `std::vector<MultiVoxelGridCovariance<pcl::PointXYZ>::Leaf>` (7.9MB).
Screenshot from 2024-10-15 18-32-27

The struct `Leaf` contains three `Eigen::Matrix3d`, and `sizeof(Leaf)` is 288, so having this for each voxel may consumes a lot of memory with large maps.

In this PR, I propose to remove `evecs_` (`Eigen::Matrix3d`) and `evals_` (`Eigen::Vector3d`) from `Leaf`. These variables are only used to calculate `cov_` and not be used directly. This will reduce `sizeof(Leaf)` from 288 to 192.

When tested with `sample-map-rosbag`, the memory usage of `std::vector<MultiVoxelGridCovariance<pcl::PointXYZ>::Leaf>` was reduced from 7.9MB to 5.2MB.

## Related links
- [Performance analysis of ndt_scan_matcher](https://tier4.atlassian.net/wiki/spaces/~6422e65c57f0c028e2f72804/pages/3344271035/ndt_scan_matcher#%E3%83%A1%E3%83%A2%E3%83%AA%E6%B6%88%E8%B2%BB%E9%87%8F) (TIER IV internal)

## How was this PR tested?
I did a regression test on https://github.com/tier4/ndt_omp and got the following result:
```
elapsed_milliseconds_ratio_mean=0.478 (current / reference)
The scores are perfectly the same.
OK
```
## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp
Comment (SakodaShintaro) [inline planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp:1246]: Are the changes in this file related to your PR?

Comment (taisa1) [reply planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp]: Sorry, it's my mistake. That part was irrelevant and I excluded it now.

**Review by SakodaShintaro — APPROVED** [review]
Looks Good To Me
I would like to wait for one more approval before merging.

**Review by YamatoAndo — APPROVED** [review]

**Pull request merged**
